test/intfuncs.jl

```diff
@@ -11,6 +11,12 @@
 
 @test typeof(gcd(Int32(3), Int32(15))) == Int32
 
+@test gcd(typemax(Int), 1) == 1
+@test gcd(-typemax(Int), 1) == 1
+@test gcd(typemin(Int), 1) == 1
+divex = sizeof(Int) == 4 ? OverflowError : InexactError
+@test_throws divex gcd(typemin(Int), typemin(Int))
```

Comment:
If we're going to the trouble of throwing something more specific than ErrorException, it's so it can be caught by type elsewhere. Hence it's important for base tests to be specific about which exception type is thrown in `@test_throws`. If the exception types are poorly designed in your opinion, that can be improved, but the current state should be tested against, carefully.